hnn_core/feed.py

```diff
@@ -7,6 +7,228 @@
 import numpy as np
 from neuron import h
 
+from .utils import check_random_state
+
+
+class Rhythmic:
+    """External inputs
+
+    Parameters
+    ----------
+    loc : str ('prox' or 'dist')
+        The location of the rhythmic input
+    params : dict
+        The params dictionary. For compatibility
+        with GUI.
+
+    Attributes
+    ----------
+    loc : str ('prox' or 'dist')
+        The location of the rhythmic input
+    t_input : array
+        The spike times.
+    """
+    def __init__(self, loc, params=None):
+
+        self.loc = loc
+        self.t_input = None
+
+        if params is not None:
+            self.set_times(tstart=params[f't0_input_{loc}'],
+                           tstop=params[f'tstop_input_{loc}'],
+                           burst_freq=params[f'f_input_{loc}'],
+                           burst_std=params[f'f_stdev_{loc}'],
+                           burst_n_spikes=params[f'events_per_cycle_{loc}'],
+                           n_bursts=params[f'repeats_{loc}'])
```

Comment:
thinking a bit more @rythorpe I think we should just focus on this method / function and do the other methods in another PR

Reply:
@jasmainak so just use the new `Evoked` and `Rhythmic` classes to initialize and store feed attributes and let the `Network` class handle setting connections for now?